iml-system-test-utils/src/docker.rs

```diff
@@ -0,0 +1,223 @@
+use crate::{iml, CheckedStatus, iml::IML_DOCKER_PATH};
+use iml_systemd;
+use iml_wire_types::Branding;
+use std::{io, thread, time, str};
+use tokio::{
+    fs::{canonicalize, File},
+    io::AsyncWriteExt,
+    process::Command,
+};
+
+pub struct DockerSetup {
+  pub use_stratagem: bool,
+  pub branding: Branding,
+}
+
+pub async fn docker() -> Result<Command, io::Error> {
+    let mut x = Command::new("docker");
+
+    let path = canonicalize(IML_DOCKER_PATH).await?;
+
+    x.current_dir(path);
+
+    Ok(x)
+}
+
+pub async fn deploy_iml_stack() -> Result<(), io::Error> {
+    iml_systemd::start_unit_and_wait("iml-docker.service".into(), 400).await?;
+
+    Ok(())
+}
+
+pub async fn get_docker_service_count() -> Result<u16, io::Error> {
+  let mut x = docker().await?;
+
+  let services = x.arg("service").arg("ls").output().await?;
+
+  let output = str::from_utf8(&services.stdout).expect("Couldn't read docker service list.");
+  let cnt: u16 = output.lines().count() as u16 - 1; // subtract the header
+
+  Ok(cnt)
+} 
+
+pub async fn remove_iml_stack() -> Result<(), io::Error> {
```

Comment:
Why do we need this fn?

The systemd unit should not yield back until there are no more services: 

https://github.com/whamcloud/integrated-manager-for-lustre/blob/fc8a24bf662790622dd13ead7c2279dd88fcd257/iml-docker.service#L16

Reply:
Right, and in most cases that works fine. There are cases, however, where running `systemctl stop iml-docker.service` immediately exits and docker services are still running. I think this happens if there is an error with one or more of the docker services. When this happens, you have to manually remove the stack and wait for all of the services to come down.

Reply:
I haven't seen this occur. We're probably better off failing the test if this happens so we can diagnose the error, rather than working around it in test.

Reply:
In that case i'll take out the extra code and it will become:
```
pub async fn remove_iml_stack() -> Result<(), iml_systemd::SystemdError> {
    iml_systemd::stop_unit("iml-docker.service".into()).await
}
```